tests/python/test_ndarray.py

```diff
@@ -1169,9 +1169,13 @@ def foo(
     assert (a[0] == vec3(3)).all()
 
 
+@pytest.mark.skipif()
 # exclude metal, because metal limited to < 30 parametrs AFAIK
 @test_utils.test(exclude=[ti.metal])
 def test_ndarray_max_num_args() -> None:
+    if platform.system() == "Darwin" and ti.lang.impl.current_cfg().arch == ti.vulkan:
+        # Mac doesn't support so many arguments, on Vulkan
+        return
```

Comment:
This is not the right pattern. Use `pytest.skip("Mac doesn't support so many arguments, on Vulkan")` instead of `return`.

Reply:
Specifically:

`@pytest.mark.skipif(sys.platform == "darwin" and ti.lang.impl.current_cfg().arch == ti.vulkan, reason="Mac doesn't support so many arguments, on Vulkan")`

Reply:
I see. Shouldn't we remove the decorator entirely then, and fully handle conditionally skipping within the function? Or does it still serve some purpose?

Reply:
leaving aside the weird empty skipif outside the func, which is probably left over from purging opengl and gles, we currently are doing the check insdie the funciton. My understanding is that alexis is propsing I replace `return` with `pytest.skip`, running as a command, inside the function?

Reply:
> I replace return with pytest.skip, running as a command, inside the function?

yes